qiskit/algorithms/minimum_eigensolvers/adapt_vqe.py

```diff
@@ -0,0 +1,280 @@
+
+    @property
+    def initial_point(self) -> Sequence[float] | None:
+        """Return the initial point."""
+        return self._solver.initial_point
+
+    @initial_point.setter
+    def initial_point(self, value: Sequence[float] | None) -> None:
+        """Set the initial point."""
+        self._solver.initial_point = value
+
+    def _compute_gradients(
+        self,
+        theta: list[float],
+        operator: OperatorBase,
+    ) -> list[tuple[complex, complex]]:
+        """
+        Computes the gradients for all available excitation operators.
+
+        Args:
+            theta: List of (up to now) optimal parameters.
+            operator: operator whose gradient needs to be computed.
+        Returns:
+            List of pairs consisting of the computed gradient and excitation operator.
+        """
+        # The excitations operators are applied later as exp(i*theta*excitation).
+        # For this commutator, we need to explicitly pull in the imaginary phase.
+        commutators = [1j * commutator(operator, exc) for exc in self._excitation_pool]
+        wave_function = self._solver.ansatz.assign_parameters(theta)
+        res = estimate_observables(self._solver.estimator, wave_function, commutators)
+        return res
+
+    @staticmethod
+    def _check_cyclicity(indices: list[int]) -> bool:
+        """
+        Auxiliary function to check for cycles in the indices of the selected excitations.
+
+        Args:
+            indices: The list of chosen gradient indices.
+
+        Returns:
+            Whether repeating sequences of indices have been detected.
+        """
+        cycle_regex = re.compile(r"(\b.+ .+\b)( \b\1\b)+")
+        # reg-ex explanation:
+        # 1. (\b.+ .+\b) will match at least two numbers and try to match as many as possible. The
+        #    word boundaries in the beginning and end ensure that now numbers are split into digits.
+        # 2. the match of this part is placed into capture group 1
+        # 3. ( \b\1\b)+ will match a space followed by the contents of capture group 1 (again
+        #    delimited by word boundaries to avoid separation into digits).
+        # -> this results in any sequence of at least two numbers being detected
+        match = cycle_regex.search(" ".join(map(str, indices)))
+        logger.debug("Cycle detected: %s", match)
+        logger.info("Alternating sequence found. Finishing.")
+        # Additionally we also need to check whether the last two numbers are identical, because the
+        # reg-ex above will only find cycles of at least two consecutive numbers.
+        # It is sufficient to assert that the last two numbers are different due to the iterative
+        # nature of the algorithm.
+        return match is not None or (len(indices) > 1 and indices[-2] == indices[-1])
+
+    def compute_minimum_eigenvalue(
+        self,
+        operator: OperatorBase,
+        aux_operators: ListOrDict[OperatorBase] | None = None,
```

Comment:
The types here should match the new MinimumEigensolver signature which is BaseOperator | PauliSumOp